src/hotspot/share/ci/ciField.cpp

```diff
@@ -91,6 +91,7 @@ ciField::ciField(ciInstanceKlass* klass, int index, Bytecodes::Code bc) :
   _signature = ciEnv::current(THREAD)->get_symbol(signature);
   _is_multifield = false;
   _is_multifield_base = false;
+  _is_transiant_multifield_base = false;
```

Comment:
Why do we still need `_is_multifield_base` in `ciField` ?

Reply:
That is one which is being used by compilers. The transient one is there for ciInstanceKlass::populate_synthetic_multifields. Both c1 and C2 use ciBytecodeStream::get_field API which creates a ciField from field reference in constant pool. It does not take into account annotations over the filed and hence populate synthetic fields operates over originally created field to create a fully populated ciMultifiled. Let me know if have a better way to address this.

Reply:
Yes, I'm wondering whether we can use one like before, by just setting `_is_multifield_base = true` only when the multifields are vectorized.

Reply:
Fixed the messy nit.